include/cutlass/gemm/device/gemm.h

```diff
@@ -480,6 +480,15 @@ class Gemm {
     cudaError_t result;
 
     int smem_size = int(sizeof(typename GemmKernel::SharedStorage));
+    
```

Comment:
the right place to do this check is in `can_implement()` above.

Also, `device::gemm_universal` is more useful than `device::gemm` now.  the former can do batch gemm, splitk in the same kernel.  `gemm_universal` can have the same smem size check.

Reply:
Thanks, I moved the block to the `can_implement` function. I wasn't aware of the `gemm_universal` template, but it seems to already have some memory checks in the `maximum_active_blocks` function. Do you think the additional checks should be added anyway?

Reply:
maximum_active_blocks is used to check the occupancy.  it does not decide if something can run or not.  if the allocated shared mem is too big, it just returns 0.